client/my-sites/media/main.jsx

```diff
@@ -44,14 +52,91 @@ module.exports = React.createClass( {
 		page( redirect );
 	},
 
+	openDetailsModal( item ) {
+		this.setState( { openedDetails: item } );
+	},
+
+	closeDetailsModal() {
+		this.setState( { openedDetails: null, editedItem: null } );
+	},
+
+	editImage() {
+		this.setState( { openedDetails: null, editedItem: this.state.openedDetails } );
+	},
+
+	onImageEditorCancel: function( imageEditorProps ) {
```

Comment:
I find it strange that the image editor's own props are being passed up back through the callback. Presumably the parent component rendering the image editor knows those props already or can determine them itself via `connect`?